scripts/ci-e2e.sh

```diff
@@ -53,7 +53,8 @@ clone_repos
 REPO_ROOT=$(realpath "$REPO_ROOT") 
 # Copy the current CAPM3 repo to the Go source directory
 rm -rf "${M3PATH}/cluster-api-provider-metal3" # To avoid 'permission denied' error when overriding .git/
-cp -R "${REPO_ROOT}" "${M3PATH}/cluster-api-provider-metal3/" 
+# TODO (Mohammed) Remove the next line since it overrides the cloned v1a5 
+# cp -R "${REPO_ROOT}" "${M3PATH}/cluster-api-provider-metal3/" 
```

Comment:
I'm afraid this will make us test the wrong code for non-upgrade tests but I'm not sure. For the other e2e tests (not upgrade) I think we checkout the code for the PR first, and we copy it here to the correct path because otherwise it would just get the main branch instead. So I think we must change this so the copy happens by default, but if it is an upgrade test we skip it.